### PR TITLE
Adding checkbox for and/or option

### PR DESCRIPTION
**Pull request by ddey2**
## Description
_Please provide a summary of the pull request and the issue it fixes. Please add necessary details, context, dependencies, explanation of when review is needed (see next section), etc._

**Fixes #(add issue number here and remove parentheses)**

[comment]: # (This project only accepts pull requests related to open issues.)
[comment]: # (If suggesting a new feature or change, please discuss it in an issue first.)
[comment]: # (If fixing a bug, there should be an issue describing it with steps to reproduce.)

## Review Time Estimate
_Please give your idea of how soon this pull request needs to be reviewed by selecting one of the options below. This can be based on the criticality of the issue at hand and/or other relevant factors._

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] Immediately
- [ ] Within a week
- [x] When possible

## Type of changes
_Please select a relevant option:_

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Other (any another change that does not fall in one of the above categories.)

## Checklist:
_Please select all applicable options:_

[comment]: # (To select your options, please put an 'x' in the all boxes that apply.)

- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My change requires updating the documentation.
- [ ] I have made necessary changes to the documentation.
- [ ] I have added tests related to my changes.
- [ ] My changes generate no new warnings.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

File: client/src/components/navHeader.vue [hunk 1]
```diff
@@ -46,6 +46,8 @@
                 placeholder="Search"
                 @keydown.enter.exact.prevent="onSubmitNavbar"
               ></b-form-input>
+              <b-form-checkbox v-model="textMatchAllButton"></b-form-checkbox>
```
Comment (valentinedwv) [inline client/src/components/navHeader.vue:49]: Title? with a javascript to say "OR" when not checked, and "AND" where checked, with a title tip describing a bit more in detail 'Selecting AND will require all provided terms to be in the result"

File: client/src/state.js [hunk 2]
```diff
@@ -576,7 +583,7 @@ export const store = _createStore({
       // })
       const resultsTemplate = _.template(SpaqlQuery, esTemplateOptions);
       //var sparql = self.state.queryTemplates[template_name]({'n': n, 'o': o, 'q': q})
-      var sparql = resultsTemplate({ n: n, o: o, q: q, rt: rt, exact: exact, minRelevance: minRelevance });
+      var sparql = resultsTemplate({ n: n, o: o, q: q, textMatchAll: match, rt: rt, exact: exact, minRelevance: minRelevance });
```
Comment (valentinedwv) [inline client/src/state.js:586]: oops, was there already and exact option, and just not implemented fully?

File: client/src/state.js [hunk 3]
```diff
@@ -58,6 +58,7 @@ export const store = _createStore({
     toolsMap: new Map(), // object id, hasConnectedTools
     q: "",
     rt: "all", // resourceType all
+    textMatchAll: false,
     searchExactMatch: false,
```
Comment (valentinedwv) [inline client/src/state.js:62]: Maybe there was a searchExactMatch... in some former codebase

File: client/src/state.js (hunk 2 above)
Comment (ddey2) [reply client/src/state.js:586]: not really. I introduced textMatchAll in state

Comment (valentinedwv) [reply client/src/state.js:586]: I know, but i think exact was in there for that purpose.
So, I think one of them can do away.

Comment (ddey2) [reply client/src/state.js]: I see what you meant. Updated the code to use searchExactMatch. I verified that it is not being used anywhere else

File: client/src/components/navHeader.vue (hunk 1 above)
Comment (ddey2) [reply client/src/components/navHeader.vue]: added. see if you can see the tooltip on your end. I can see it may 20% of the time.

**Review by valentinedwv — APPROVED** [review]
Added a css hack... not good at these, but color could be fixed.

approved

**Comment by ddey2** [comment]
UI with the new flip switch
Screenshot 2024-09-30 at 10 14 23 AM
Screenshot 2024-09-30 at 9 57 49 AM

File: client/package.json [hunk 4]
```diff
@@ -46,6 +46,7 @@
     "vue-router": "^4.2.5",
     "vue-scrollbox": "^1.0.1",
     "vue-select": "4.0.0-beta.6",
+    "vue-toggles": "1.1.4",
```
Comment (valentinedwv) [inline client/package.json:49]: use latest version. branch i developed worked

File: client/src/state.js [hunk 5]
```diff
@@ -562,6 +562,8 @@ export const store = _createStore({
       if (resourceType !== undefined && resourceType !== "all") {
         rt = this.state.resourceTypeList.get(resourceType);
       }
+      if (exact == undefined)
+        exact = this.state.searchExactMatch;
```
Comment (valentinedwv) [inline client/src/state.js:566]: Does this default to True or false. can we set it with an environment variable

**Review by valentinedwv — COMMENTED** [review]
Waiting for a deployment of the UI for testing.

File: client/src/state.js (hunk 5 above)
Comment (ddey2) [reply client/src/state.js:566]: Default is always False. 

I can try to set it with an environment variable but I think the default should be False. In this case, False means OR and that will get more search results.

**Review by valentinedwv — COMMENTED** [review]
The spinner when query from the search page is running  seems to happen after the query has returned

do a query on https://geocodes.earthcube.org, 
say steens, 
then set the toggle to and the search do water nitrogen

the do one on https://geocodes.geocodes-aws.earthcube.org/

**Pull request merged**